### ISSUE TITLE
7.1 Release Prep: Unit Tests Failing

### ISSUE THREAD

**Issue by nyghtowl** [deeplearning4j/deeplearning4j#2391]
Core:
CustomPreprocessorTest.testCustomPreprocessor:44 MyCustomPreprocessor: not registered with NeuralNetConfiguration mapper
  TestVariableLengthTSCG.testVariableLengthSimple:91 expected:<0.46309762478286115> but was:<0.4630976247828612>
  TestCustomLayers.testCustomOutputLayerMLN:159 CustomOutputLayer: not registered with NeuralNetConfiguration mapper
  TestCustomLayers.testJsonMultiLayerNetwork:67 CustomLayer: not registered with NeuralNetConfiguration mapper
  ~~BatchNormalizationTest.checkMeanVarianceEstimate:585 arrays first differed at element [1]; expected:<0.5> but was:<0.42191187>~~
  ~~BatchNormalizationTest.checkMeanVarianceEstimateCNN:646 arrays first differed at element [0]; expected:<0.5> but was:<0.48279557>~~
~~ GravesBidirectionalLSTMTest.testConvergence:495 null~~
  TestVariableLengthTS.testVariableLengthSimple:91 expected:<0.46309762478286115> but was:<0.4630976247828612>

UI:
  FlowIterationListenerTest.testCGModelInfo1:185 » ArrayIndexOutOfBounds 0
  FlowIterationListenerTest.testMLNModelInfo1:155 » NullPointer
  TestFlowListener.testUICG:144 » ArrayIndexOutOfBounds 0

Spark:
  Run 1: TestDataVecDataSetFunctions>BaseSparkTest.before:60->BaseSparkTest.getContext:94 » IllegalState
  Run 2: TestDataVecDataSetFunctions>BaseSparkTest.after:77 » NullPointer

**Comment by saudet** [comment]
Seeing similar failures, but nothing really new I think except the Akka stuff:
```
Failed tests: 
  CustomPreprocessorTest.testCustomPreprocessor:44 MyCustomPreprocessor: not registered with NeuralNetConfiguration mapper
  TestGraphNodes.testL2Node:379 expected: org.nd4j.linalg.cpu.nativecpu.NDArray<[[   �,    �],
 [   �,    �],
 [   �,    �],
 [   �,    �],
 [   �,    �]]> but was: org.nd4j.linalg.cpu.nativecpu.NDArray<[[   �,    �],
 [   �,    �],
 [   �,    �],
 [   �,    �],
 [   �,    �]]>
  TestCustomLayers.testCustomOutputLayerMLN:159 CustomOutputLayer: not registered with NeuralNetConfiguration mapper
  TestCustomLayers.testJsonMultiLayerNetwork:67 CustomLayer: not registered with NeuralNetConfiguration mapper
  BatchNormalizationTest.checkMeanVarianceEstimate:585 arrays first differed at element [1]; expected:<0.5> but was:<0.42191187>
  BatchNormalizationTest.checkMeanVarianceEstimateCNN:646 arrays first differed at element [0]; expected:<0.5> but was:<0.48279557>
  GravesBidirectionalLSTMTest.testConvergence:495 null
Tests in error: 
  ComputationGraphConfigurationTest.testConfigurationWithRuntimeJSONSubtypes:212 » Runtime
  ActivationLayerTest.testAutoEncoderActivationLayer:128 » IllegalState Set pret...
  ConvolutionLayerTest.testCNNMLNPretrain:382 » NullPointer
  TestCustomLayers.testCustomOutputLayerCG:231 » Runtime org.nd4j.shade.jackson....
  TestCustomLayers.testJsonComputationGraph:110 » Runtime org.nd4j.shade.jackson...
  MultiLayerTest.testTranspose:512 » Runtime unable to construct transposed laye...

Tests run: 479, Failures: 7, Errors: 6, Skipped: 8
```

```
Failed tests: 
  TestSparkComputationGraph.testSeedRepeatability:263 expected:<[0.31, 0.92, 1.26, 0.02, -0.72, 0.15, -0.07, 0.00, -0.49, 0.56, 0.17, -0.10, 0.63, -0.74, 0.00, -0.11, 0.06, 0.13, -0.12, -0.17, -0.08, 1.09, 1.19, -0.25, -1.19, 0.32, -0.31, -0.32, -0.62, 0.95, 0.40, -0.15, -0.25, -0.29, -0.22]> but was:<[0.32, 0.92, 1.26, 0.02, -0.71, 0.16, -0.06, 0.03, -0.49, 0.56, 0.21, -0.10, 0.73, -0.64, 0.10, -0.07, 0.06, 0.14, -0.12, -0.11, -0.08, 1.09, 1.30, -0.25, -1.19, 0.32, -0.29, -0.32, -0.62, 0.95, 0.52, -0.15, -0.25, -0.29, -0.22]>
  TestSparkMultiLayerParameterAveraging.testSeedRepeatability:749 expected:<[0.32, 0.91, 1.24, 0.02, -0.68, 0.18, -0.04, 0.04, -0.39, 0.70, 0.31, 0.04, 0.76, -0.60, 0.13, -0.01, 0.05, 0.16, 0.02, -0.05, -0.13, 1.14, 1.25, -0.29, -1.19, 0.32, -0.26, -0.32, -0.62, 0.95, 0.26, -0.15, -0.30, -0.29, -0.22]> but was:<[0.32, 0.92, 1.24, 0.02, -0.73, 0.14, -0.09, 0.02, -0.39, 0.70, 0.31, 0.04, 0.66, -0.71, 0.03, -0.12, 0.05, 0.12, 0.02, -0.15, -0.13, 1.14, 1.29, -0.29, -1.19, 0.32, -0.31, -0.32, -0.62, 0.95, 0.35, -0.15, -0.30, -0.29, -0.22]>
```

```
Results :

Tests in error: 
  PipelineTest.testPipeline:52 » FailedToCreateRoute Failed to create route rout...
  Dl4jServingRouteTest.testServingRoute:170 » FileNotFound File 'tmpdir/tmp.txt'...

Tests run: 2, Failures: 0, Errors: 2, Skipped: 0
```
```

Tests in error: 
  TestSerialization.testModelSerde:48 » Runtime java.lang.ClassNotFoundException...
  FlowIterationListenerTest.testCGModelInfo1:183 » ClassNotFound akka.cluster.Cl...
  FlowIterationListenerTest.testMLNModelInfo1:153 » ClassNotFound akka.cluster.C...
  TestFlowListener.testUI:78 » ClassNotFound akka.cluster.ClusterActorRefProvide...
  TestFlowListener.testUICG:141 » ClassNotFound akka.cluster.ClusterActorRefProv...
  TestConvolutionalListener.testUI:75 » ClassNotFound akka.cluster.ClusterActorR...

Tests run: 13, Failures: 0, Errors: 6, Skipped: 3
```

**Comment by agibsonccc** [comment]
Looks like all classpath related again

**Comment by AlexDBlack** [comment]
Not all, the L2 vertex is RNG related. Possibly also the batch norm stuff too... Looking into it now.

**Comment by AlexDBlack** [comment]
L2 vertex and BatchNormalizationTest tests are now passing after this fix: https://github.com/deeplearning4j/libnd4j/pull/337

Custom layer stuff is passing in intellij (that's classpath stuff - reflections library not finding classes probably).

Also seeing the following, which I'll look into:
ActivationLayerTest.testAutoEncoderActivationLayer:128 » IllegalState Set pret...
ConvolutionLayerTest.testCNNMLNPretrain:382 » NullPointer
MultiLayerTest.testTranspose:512 » Runtime unable to construct transposed laye...

**Comment by AlexDBlack** [comment]
After merging this PR: I'm only seeing the native library issues (on Windows) + nd4j-native.
https://github.com/deeplearning4j/deeplearning4j/pull/2395

**Comment by nyghtowl** [comment]
Still seeing above except for those that I've noted

**Comment by AlexDBlack** [comment]
Custom layers and UI issues should be fixed.
I haven't previously seen issues like TestVariableLengthTSCG.testVariableLengthSimple or Spark or Windows (native or cuda).

**Comment by AlexDBlack** [comment]
These should be fixed - closing this.
